plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,811 @@
+#!/usr/bin/python
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://wwww.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: s3_object_info
+version_added: 5.0.0
+short_description: Gather information about objects in S3
+description:
+    - Describes objects in S3.
+    - Compatible with AWS, DigitalOcean, Ceph, Walrus, FakeS3 and StorageGRID (only supports list_keys currently).
```

Comment:
See https://github.com/ansible-collections/amazon.aws/pull/977#pullrequestreview-1106205983

We're kinda stuck because the s3_object (aws_s3) module has 'supported' them since the start of this collection, and part of the reason to create s3_object_info is so that we can move the 'info' functionality out of that module.

Reply:
Thank you @tremble.